File: Sources/ControlVCore/Models/FeedbackSubmission.swift
import Foundation

public enum FeedbackCategory: String, Codable, CaseIterable, Identifiable {
    case bug
    case idea
    case praise
    case other

    public var id: String { rawValue }

    public var label: String {
        switch self {
        case .bug: return "Bug"
        case .idea: return "Idea"
        case .praise: return "Love it"
        case .other: return "Other"
        }
    }
}

/// One piece of in-app feedback, exactly as sent to `submit-feedback`.
public struct FeedbackSubmission: Codable, Equatable {
    public var rating: Int?
    public var category: FeedbackCategory
    public var message: String
    public var contactEmail: String?
    public var installID: String
    public var sessionToken: String?
    public var appVersion: String
    public var platform: String

    public init(
        rating: Int?,
        category: FeedbackCategory,
        message: String,
        contactEmail: String?,
        installID: String,
        sessionToken: String?,
        appVersion: String,
        platform: String = "macos"
    ) {
        self.rating = rating
        self.category = category
        self.message = message
        self.contactEmail = contactEmail
        self.installID = installID
        self.sessionToken = sessionToken
        self.appVersion = appVersion
        self.platform = platform
    }
}

File: windows/ControlV.Core/Models/FeedbackSubmission.cs
using System.Text.Json.Serialization;

namespace ControlV.Core.Models;

public enum FeedbackCategory { Bug, Idea, Praise, Other }

public static class FeedbackCategories
{
    public static string RawValue(this FeedbackCategory c) => c.ToString().ToLowerInvariant();
    public static string Label(this FeedbackCategory c) => c switch
    {
        FeedbackCategory.Bug => "Bug", FeedbackCategory.Idea => "Idea", FeedbackCategory.Praise => "Love it", _ => "Other",
    };
}

/// Exactly the JSON `submit-feedback` expects (same as the Mac client).
public sealed class FeedbackSubmission
{
    [JsonPropertyName("rating")] public int? Rating { get; set; }
    [JsonPropertyName("category")] public string Category { get; set; } = FeedbackCategory.Idea.RawValue();
    [JsonPropertyName("message")] public string Message { get; set; } = "";
    [JsonPropertyName("contactEmail")] public string? ContactEmail { get; set; }
    [JsonPropertyName("installID")] public string InstallId { get; set; } = "";
    [JsonPropertyName("sessionToken")] public string? SessionToken { get; set; }
    [JsonPropertyName("appVersion")] public string AppVersion { get; set; } = "";
    [JsonPropertyName("platform")] public string Platform { get; set; } = "windows";
}
